openmc/volume.py

```diff
@@ -0,0 +1,204 @@
+from collections import Iterable, Mapping
+from numbers import Real, Integral
+from xml.etree import ElementTree as ET
+
+import numpy as np
+import pandas as pd
+
+from openmc import Cell, Union
+import openmc.checkvalue as cv
+
+
+class VolumeCalculation(object):
+    """Stochastic volume calculation specifications and results.
+
+    Parameters
+    ----------
+    cells : Iterable of Cell
+        Cells to find volumes of
```

Comment:
Think ahead about possibly having the ability to calculate volumes for universes, materials, or meshes, do you think it would be worthwhile to generalize this now with something akin to what is done in the mgxs module (e.g. replacing `cells` with `domains` and `domain_type`)? Then just have a `NotImplementedError` for the other domain types.